lib/text/speech_to_text.js

```diff
@@ -0,0 +1,597 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.text.SpeechToText');
+
+goog.require('goog.asserts');
+goog.require('shaka.log');
+goog.require('shaka.util.ArrayUtils');
+goog.require('shaka.util.Dom');
+goog.require('shaka.util.EventManager');
+goog.require('shaka.util.FakeEvent');
+goog.require('shaka.util.IReleasable');
+goog.require('shaka.util.ManifestParserUtils');
+goog.require('shaka.util.Lazy');
+goog.require('shaka.util.Timer');
+goog.requireType('shaka.Player');
+
+
+/**
+ * @implements {shaka.util.IReleasable}
+ */
+shaka.text.SpeechToText = class {
+  /**
+   * @param {shaka.Player} player
+   */
+  constructor(player) {
+    /** @private {?shaka.Player} */
+    this.player_ = player;
+
+    /** @private {?shaka.extern.SpeechToTextConfiguration} */
+    this.config_ = null;
+
+    /** @private {!shaka.util.EventManager} */
+    this.eventManager_ = new shaka.util.EventManager();
+
+    /** @private {boolean} */
+    this.supported_ =
+        shaka.text.SpeechToText.isMediaStreamTrackSupported.value();
+
+    /** @type {HTMLElement} */
+    this.textContainer_ = this.getTextContainer_();
+
+    /** @private {boolean} */
+    this.enabled_ = false;
+
+    /** @private {?ChromeSpeechRecognition} */
+    this.recognition_ = null;
+
+    /** @private {?Translator} */
+    this.translator_ = null;
+
```

Comment:
Is it possible to move translation feature to separate file/class that would be used by this module?

If that would add more complexity than needed, let's leave it as-is.